### PR TITLE
Add support for specifying the viewbox of path elements

### PR DESCRIPTION
**Pull request by tronical**

### CODE REVIEW

File: docs/builtin_elements.md [hunk 1]
```diff
@@ -227,6 +227,10 @@ accordingly.
 * **`stroke-width`** (*length*): The width of the outline.
 * **`width`** (*length*): If non-zero, the path will be scaled to fit into the specified width.
 * **`height`** (*length*): If non-zero, the path will be scaled to fit into the specified height.
+* **`viewbox-x`**/**`viewbox-y`**/**`viewbox-width`**/**`viewbox-height`** (*float*) These four properties allow
+  defining the position and size of the viewport of the path in path coordinates. In the rendered output, the
+  If the `viewport-width` or `viewport-height` is less or equal than zero, the viewbox properties are ignored
```
Comment (ogoffart) [inline docs/builtin_elements.md:232]: viewport?

File: examples/iot-dashboard/iot-dashboard.60 [hunk 2]
```diff
@@ -118,6 +118,11 @@ PieChartPainted := Rectangle {
         height: 100%;
         stroke-width: parent.width / 6;
         stroke: #aaaaaa40;
+        viewbox-x: 0;
+        viewbox-y: 0;
```
Comment (ogoffart) [inline examples/iot-dashboard/iot-dashboard.60:122]: do we need to specify the 0?

**Review by ogoffart — COMMENTED** [review]
Great!

Also missing an entry in the changelog

**Comment by ogoffart** [comment]
What happens if we draw outside of the viewbox ?

File: docs/builtin_elements.md (hunk 1 above)
Comment (tronical) [reply docs/builtin_elements.md:232]: Ooops

File: examples/iot-dashboard/iot-dashboard.60 (hunk 2 above)
Comment (tronical) [reply examples/iot-dashboard/iot-dashboard.60:122]: Indeed that’s not necessary

**Comment by tronical** [comment]
Ah I forgot to git add the Changelog bit.

I think I need to amend the change to add clipping of the viewbox intersects with the bounding box of the path.

**Comment by ogoffart** [comment]
> I think I need to amend the change to add clipping of the viewbox intersects with the bounding box of the path

Maybe only if clip is true.
We probably need to special case  clip on path to put it as a parent instead of as a child

**Comment by tronical** [comment]
Good idea to re-use the `clip` property. I'll do that next.

**Review by ogoffart — APPROVED** [review]

File: docs/builtin_elements.md [hunk 3]
```diff
@@ -231,6 +231,9 @@ accordingly.
   defining the position and size of the viewport of the path in path coordinates. In the rendered output, the
   If the `viewbox-width` or `viewbox-height` is less or equal than zero, the viewbox properties are ignored
   and instead the bounding rectangle of all path elements are used to define the view port.
+* **`clip`** (*bool*): By default, when a path has a view box defined and the elements render outside of it, they are still
```
Comment (ogoffart) [inline docs/builtin_elements.md:234]: Is it necessary to document this here since it is a property of all items?

Comment (ogoffart) [reply docs/builtin_elements.md:234]: ah, it is actually not a property on all items, only on rectangle.  then it make sense

**Review by ogoffart — APPROVED** [review]

**Pull request merged**